archivist/composite.py

```diff
@@ -0,0 +1,99 @@
+"""Composite interface
+
+   Access to various utility methods that access various endpoints.
+
+   The user is not expected to use this class directly. It is an attribute of the
+   :class:`Archivist` class.
+
+   For example instantiate an Archivist instance and execute the methods of the class:
+
+   .. code-block:: python
+
+      with open(".auth_token", mode="r", encoding="utf-8") as tokenfile:
+          authtoken = tokenfile.read().strip()
+
+      # Initialize connection to Archivist
+      arch = Archivist(
+          "https://app.rkvst.io",
+          authtoken,
+      )
+      asset = arch.composite.......(...)
+
+"""
+
+from collections import Counter
+from logging import getLogger
+
+# pylint:disable=unused-import      # To prevent cyclical import errors forward referencing is used
+# pylint:disable=cyclic-import      # but pylint doesn't understand this feature
+from . import archivist as type_helper
+
+
+LOGGER = getLogger(__name__)
+
+
+class _CompositeClient:
+    """CompositeClient
+
+    Access to various composite methods,
+    accessed as an attribute of the Archivist class.
+
+    Args:
+        archivist (Archivist): :class:`Archivist` instance
+
+    These mthods are not unittested and provided as a convenience.
+    """
+
+    def __init__(self, archivist: "type_helper.Archivist"):
+        self._archivist = archivist
+
+    def __str__(self) -> str:
+        return f"CompositeClient({self._archivist.url})"
+
+    def estate_info(self):
+        """
+        Evaluate health of the various assets and events in the sytem
+
+        The report is emitted using LOGGER.info statements
+        """
+        # for around 550 events and 250 assets this can take about a 90s...
```

Comment:
Does it not use the quick counting method (the COUNT header)?

Reply:
yes it does for the simple case - the cross-checking code requires the list() calls but if you think this is no longer needed than can be simplified greatly...